core/base/connectedComponents/ConnectedComponents.h

```diff
@@ -0,0 +1,159 @@
+/// \ingroup base
+/// \class ttk::ConnectedComponents
+/// \author Jonas Lukasczyk <jl@jluk.de>
+/// \date 01.02.2019
+///
+/// \brief TTK %connectedComponents processing package.
+///
+/// This filter consumes a scalar field with a feature mask and computes for
+/// each edge connected group of vertices with a non-background mask value a
+/// so-called connected component via flood-filling, where the backgroud is
+/// masked with values smaller-equal zero. The computed components store the
+/// size, seed, and center of mass of each component. The flag
+/// UseSeedIdAsComponentId controls if the resulting segmentation is either
+/// labeled by the index of the component, or by its seed location (which can be
+/// used as a deterministic component label).
+
+#pragma once
+
+// base code includes
+#include <Debug.h>
+#include <Triangulation.h>
+
+typedef ttk::SimplexId TID;
+
+namespace ttk {
+  class ConnectedComponents : virtual public Debug {
+
+  private:
+    static const int UNLABELED{-2};
+    static const int IGNORE{-1};
+
+  public:
+    struct Component {
+      int seed = -1;
+      float center[3]{0, 0, 0};
+      float size{0};
+    };
+
+    ConnectedComponents() {
+      this->setDebugMsgPrefix("ConnectedComponents");
+    };
+    ~ConnectedComponents(){};
```

Comment:
should be virtual if class is not marked as final

Reply:
The issue with missing virtual destructors is also present in a lot of ttk filters (including hello world). I will fix this here but this should also be addressed in an own PR.